src/main/java/org/mvplugins/multiverse/core/configuration/functions/NodeSerializer.java

```diff
@@ -0,0 +1,26 @@
+package org.mvplugins.multiverse.core.configuration.functions;
+
+/**
+ * A function that serializes and deserializes objects to and from YAML.
+ *
+ * @param <T>   The type of the object to serialize and deserialize.
+ */
+public interface NodeSerializer<T> {
```

Comment:
I wonder if there's benefit to breaking this into 2 separate functional interfaces for serialization and deserialization...

Reply:
well you cant really have a serialization without a deserialization

Reply:
I was just thinking more for the fact that then they could be implemented with lambdas.